dashboard/src/modules/components/TableComponent/index.jsx

```diff
@@ -47,9 +48,11 @@ const TableWithFavorite = () => {
   const dispatch = useDispatch();
 
   useEffect(() => {
-    dispatch(fetchPublicDatasets());
-    dispatch(getFavoritedDatasets());
-  }, [dispatch]);
+    if (Object.keys(endpoints).length > 0) {
+      dispatch(fetchPublicDatasets());
+      dispatch(getFavoritedDatasets());
+    }
+  }, [dispatch, endpoints]);
```

Comment:
Do we need to check for the existence of the endpoint list here?  That is, can we actually get this far if we cannot load the endpoints?

(If we can and we do, then aren't there a bunch of other places which need this protection, too?  This is the first one that I've noted in this PR.)